yt/frontends/amrvac/tests/test_outputs.py

```diff
@@ -21,118 +26,113 @@
-    ds = data_dir_load(blastwave_cylindrical_3D)
-    for test in small_patch_amr(ds, _get_fields_to_check(ds)):
-        test_blastwave_cylindrical_3D.__name__ = test.description
-        yield test
-
-
-@requires_ds(khi_cartesian_2D)
-def test_khi_cartesian_2D():
-    ds = data_dir_load(khi_cartesian_2D)
-    for test in small_patch_amr(ds, _get_fields_to_check(ds)):
-        test_khi_cartesian_2D.__name__ = test.description
-        yield test
-
-
-@requires_ds(khi_cartesian_3D)
-def test_khi_cartesian_3D():
-    ds = data_dir_load(khi_cartesian_3D)
-    for test in small_patch_amr(ds, _get_fields_to_check(ds)):
-        test_khi_cartesian_3D.__name__ = test.description
-        yield test
-
-
-@requires_ds(jet_cylindrical_25D)
-def test_jet_cylindrical_25D():
-    ds = data_dir_load(jet_cylindrical_25D)
-    for test in small_patch_amr(ds, _get_fields_to_check(ds)):
-        test_jet_cylindrical_25D.__name__ = test.description
-        yield test
-
-
-@requires_ds(riemann_cartesian_175D)
-def test_riemann_cartesian_175D():
-    ds = data_dir_load(riemann_cartesian_175D)
-    for test in small_patch_amr(ds, _get_fields_to_check(ds)):
-        test_riemann_cartesian_175D.__name__ = test.description
-        yield test
-
-
-@requires_ds(rmi_cartesian_dust_2D)
-def test_rmi_cartesian_dust_2D():
-    # dataset with dust fields
-    ds = data_dir_load(rmi_cartesian_dust_2D)
-    for test in small_patch_amr(ds, _get_fields_to_check(ds)):
-        test_rmi_cartesian_dust_2D.__name__ = test.description
-        yield test
+    # return [fields, field_ids]
+    return [ds, fields]
+
+
+def get_pairs():
+    f_list = [_get_fields_to_check(fname) for fname in ds_list]
+    pairs = [(i[0], f) for i in f_list for f in i[1]]
+    return pairs
+
+
+@pytest.mark.answer_test
+class TestAMRVAC:
+    answer_file = None
+    saved_hashes = None
+
+    @requires_file(khi_cartesian_2D)
+    def test_AMRVACDataset(self):
+        assert isinstance(utils.data_dir_load(khi_cartesian_2D), AMRVACDataset)
+
+    @utils.requires_ds(blastwave_cartesian_3D)
+    def test_domain_size(self):
+        # "Check for correct box size, see bw_3d.par"
+        ds = utils.data_dir_load(blastwave_cartesian_3D)
+        for lb in ds.domain_left_edge:
+            assert int(lb) == 0
+        for rb in ds.domain_right_edge:
+            assert int(rb) == 2
+        for w in ds.domain_width:
+            assert int(w) == 2
+
+    @requires_file(blastwave_cartesian_3D)
+    def test_grid_attributes(self):
+        # "Check various grid attributes"
+        ds = utils.data_dir_load(blastwave_cartesian_3D)
+        grids = ds.index.grids
+        assert ds.index.max_level == 2
+        for g in grids:
+            assert isinstance(g, AMRVACGrid)
+            assert isinstance(g.LeftEdge, YTArray)
+            assert isinstance(g.RightEdge, YTArray)
+            assert isinstance(g.ActiveDimensions, np.ndarray)
+            assert isinstance(g.Level, (np.int32, np.int64, int))
```

Comment:
I don't know why such a variability is allowed in typing here, I don't think there's a good reason for it. I'd rather have that test fail so we can fix it. I think the correct type here would be either `int` or `np.int8`.
Note that when we really don't care about the exact type but need to check it's a integer of some sort (for instance when it's user provided data), you can use `isinstance(var, (int, np.integer))`